code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -196,7 +196,7 @@
 
 	else //Something like a glass. Player probably wants to transfer TO it.
 
-		if(!istype(target, /obj/item/weapon/reagent_containers)) //Not a reagent container, we most likely shouldn't be dealing with it here !
+		if(!istype(target, /obj/item/weapon/reagent_containers) && !istype(target, /obj/machinery/microwave)) //Not a reagent container or microwave, we most likely shouldn't be dealing with it here !

```

Comment:
I suggest refactoring this into a proc that's called on the target to return if we should try to interact with it or just slap it/splash it.